disnake/ext/commands/interaction_bot_base.py

```diff
@@ -1364,3 +1369,66 @@ async def on_application_command_autocomplete(
         self, interaction: ApplicationCommandInteraction
     ) -> None:
         await self.process_app_command_autocompletion(interaction)
+
+    def _parse_sync_warning(self, exception: Exception, commands: List[ApplicationCommand]) -> str:
+        if not isinstance(exception, HTTPException) or exception.status != 400:
+            return str(exception)
+
+        def _parse_options(
+            message: str,
+            key: str,
+            value: dict,
+            options: List[Option | OptionChoice],
+            indent_level: int = 2,
+            metadata: str = "",
+        ):
+
+            int_key = None
+            new_metadata = ""
+            try:
+                int_key = int(key)
+                key = options[int_key].name
+                new_metadata = f" ({metadata} {int_key})"
+            except ValueError:
+                pass
+
+            message += f"{' ' * indent_level}{key}{new_metadata}:\n"
+            indent_level += 2
+
+            if "_errors" in value:
+                message += f"{' ' * indent_level}{_flatten_error_dict({key: value}).get(key)}\n"
+                return message
+
+            if int_key and getattr(options[int_key], "options", []):
+                metadata = "Option"
+                options = options[int_key].options
+            elif int_key and getattr(options[int_key], "choices", []):
+                metadata = "Choice"
+                options = options[int_key].choices
+
+            for k, v in value.items():
+                message = _parse_options(message, k, v, options, indent_level, metadata)
+
+            return message
+
+        try:
+            sync_warnings = []
+            for command_num, error in exception.errors.items():
+                command = commands[int(command_num)]
```

Comment:
`_parse_options` catches `ValueError`s on the index int cast, something similar could be done here (or maybe even merge them into one general-purpose snippet, if possible? not that important.).
Additionally, an [`isdecimal`](https://docs.python.org/3.11/library/stdtypes.html#str.isdecimal) check would be a bit more clean than catching errors; if that check fails, it should just fall back to the default behavior of using `_flatten_error_dict` for that entry.